pkg/agent/client.go

```diff
@@ -50,6 +50,14 @@ func (c *connContext) cleanup() {
 	c.cleanOnce.Do(c.cleanFunc)
 }
 
+func (c *connContext) send(msg []byte) {
+	defer func() {
+		// Handles the race condition where we write to a closed channel
+		recover()
```

Comment:
worth a warning log when this happens? Ideally would include connID.

Reply:
Maybe an info log? We kind of know this will happen in normal execution flows.